### PR TITLE
Remove iceberg-pig

### PR DESCRIPTION
**Pull request by jbonofre**
Following #11379 this PR completely removes iceberg-pig (for Iceberg 1.8.0).

### CODE REVIEW

**Comment by manuzhang** [comment]
When I try removing Hive 2 in  https://github.com/apache/iceberg/pull/10996, I find both Hive and Pig are referenced in https://github.com/apache/iceberg/blob/main/mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java#L350. Shall we remove them as well?

**Comment by jbonofre** [comment]
@manuzhang good catch ! I think it makes sense to remove from there. I will update the PR, thanks !

**Comment by jbonofre** [comment]
@manuzhang I did a more "agressive" pass on removing pig 😄

**Comment by ajantha-bhat** [comment]
we can keep it in Draft since it is for 1.8.0

**Comment by jbonofre** [comment]
@ajantha-bhat done

**Comment by nastra** [comment]
please also update all of the yaml files for GH actions

**Comment by Fokko** [comment]
@jbonofre I think we're ready :)

**Comment by jbonofre** [comment]
@Fokko yup changing the PR status :)

**Comment by jbonofre** [comment]
@Fokko the PR is ready to review (I added additional cleanups).

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -75,8 +75,7 @@ netty-buffer = "4.1.114.Final"
 netty-buffer-compat = "4.1.114.Final"
 object-client-bundle = "3.3.2"
 orc = "1.9.4"
-parquet = "1.13.1"
-pig = "0.17.0"
+parquet = "1.14.3"
```
Comment (Fokko) [inline gradle/libs.versions.toml:78]: This looks like a merge conflict gone wrong. We've fallen back to 1.13.1 due to a bug in the dictionary encoding, we have https://github.com/apache/iceberg/pull/11502 to bump to `1.14.4`

Comment (jbonofre) [reply gradle/libs.versions.toml]: Fixed.

**Review by manuzhang — APPROVED** [review]

**Review by nastra — APPROVED** [review]

**Pull request merged**